### PR TITLE
Send rejected comment notification

### PR DESCRIPTION
**Pull request by tiagograf-AOT**
*Issue #:*
https://github.com/bcgov/met-public/issues/886

*Description of changes:*
Send rejected comment notification

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the met-public license (Apache 2.0).

### CODE REVIEW

File: met-web/src/components/comments/admin/review/CommentReview.tsx [hunk 1]
```diff
@@ -59,7 +59,14 @@ const CommentReview = () => {
     const handleSave = async () => {
         setIsSaving(true);
         try {
-            await reviewComments({ submission_id: Number(submissionId), status_id: review });
+            await reviewComments({
+                submission_id: Number(submissionId),
+                status_id: review,
+                has_personal_info: true,
```
Comment (tiagograf-AOT) [inline met-web/src/components/comments/admin/review/CommentReview.tsx:65]: static code for now until front-end is done.

File: met-api/src/met_api/services/submission_service.py [hunk 2]
```diff
@@ -66,18 +74,34 @@ def _validate_fields(submission):
...
         valid_statuses = [status.id for status in CommentStatus.get_comment_statuses()]
 
         if not status_id or status_id == Status.Pending.value or status_id not in valid_statuses or not user:
-            raise ValueError('Invalid review')
+            raise ValueError('Invalid review status.')
+
+        if status_id == Status.Rejected.value and \
+           has_personal_info is not True and \
+           has_profanity is not True and \
+           has_threat is not True and \
```
Comment (saravanpa-aot) [inline met-api/src/met_api/services/submission_service.py:94]: not any () might be useful to write this in a better way ;but good as is as well..

File: met-api/migrations/versions/ffac8f5b4288_add_reject_reason.py [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+
+# revision identifiers, used by Alembic.
+revision = 'ffac8f5b4288'
+down_revision = '46490dc02be5'
+branch_labels = None
+depends_on = None
+
+
+def upgrade():
+    op.add_column('submission', sa.Column('has_personal_info', sa.Boolean(), nullable=True))
```
Comment (saravanpa-aot) [inline met-api/migrations/versions/ffac8f5b4288_add_reject_reason.py:20]: if you want to write with less characters , i will go for has_PI , has_abuse etc..but sine you already done it , let it eb like this..
I am thinking of either just **rejected_reason** or **rejected_reason_text** than **rejected_reason_other** ;we might forget down the line whats the **other** for...just **rejected_reason** itself is pretty good i feel ;but no need to change if you dont feel so

**Review by saravanpa-aot — APPROVED** [review]
You got a lot of work done in this PR

**Comment by jadmsaadaot** [comment]
Great work handling duplication in the unit tests and separating the files. Will have a few merge conflicts with my PR that I will handle once this is merged. This should merge first

File: met-web/src/components/comments/admin/review/CommentReview.tsx [hunk 4]
```diff
@@ -36,6 +55,11 @@ const CommentReview = () => {
 
             const fetchedSubmission = await getSubmission(Number(submissionId));
             setSubmission(fetchedSubmission);
+            setHasOtherReason(!!fetchedSubmission.rejected_reason_other);
```
[the remark this answers is not in the document]
Comment (tiagograf-AOT) [reply met-web/src/components/comments/admin/review/CommentReview.tsx:58]: I'd have to spent some precious time at this point to implement it, we can have a future work on this, i'll keep that in mind for any new work

File: met-api/src/met_api/services/submission_service.py (hunk 2 above)
Comment (tiagograf-AOT) [reply met-api/src/met_api/services/submission_service.py:94]: updated

File: met-api/src/met_api/services/submission_service.py [hunk 5]
```diff
@@ -66,18 +74,34 @@ def _validate_fields(submission):
...
+        has_threat = reasons.get('has_threat', None)
+        rejected_reason_other = reasons.get('rejected_reason_other', None)
 
         valid_statuses = [status.id for status in CommentStatus.get_comment_statuses()]
 
         if not status_id or status_id == Status.Pending.value or status_id not in valid_statuses or not user:
-            raise ValueError('Invalid review')
+            raise ValueError('Invalid review status.')
+
+        if status_id == Status.Rejected.value and \
```
[the remark this answers is not in the document]
Comment (tiagograf-AOT) [reply met-api/src/met_api/services/submission_service.py]: I split the validation to a separated method, I think marshmallow might make this validation a bit trickier to understand, but we can consider it in the future, I won't implement it as part of this PR

File: met-api/src/met_api/config.py [hunk 6]
```diff
@@ -96,6 +96,8 @@ class _Config():  # pylint: disable=too-few-public-methods
     # Email Service
     VERIFICATION_EMAIL_SUBJECT = os.getenv('VERIFICATION_EMAIL_SUBJECT', '{engagement_name} - Survey link')
     VERIFICATION_EMAIL_TEMPLATE_ID = os.getenv('VERIFICATION_EMAIL_TEMPLATE_ID')
+    REJECTED_EMAIL_TEMPLATE_ID = os.getenv('REJECTED_EMAIL_TEMPLATE_ID')
+    REJECTED_EMAIL_SUBJECT = os.getenv('REJECTED_EMAIL_SUBJECT', '{engagement_name} - About your Comments')
```
[the remark this answers is not in the document]
Comment (tiagograf-AOT) [reply met-api/src/met_api/config.py:100]: updated

**Review by VineetBala-AOT — APPROVED** [review]

**Pull request merged**